Sources/SentryDistribution/Updater.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import UIKit
+
+public final class Updater {
+  // MARK: - Public
+  public static let shared = Updater()
+
+  /// Checks if there is an update available for the app, based on the provided `params`.
+  ///
+  ///
+  /// - Parameters:
+  ///   - params: A `CheckForUpdateParams` object.
+  ///   - completion: A closure that is called with the result of the update check.
+  ///
+  /// - Example:
+  /// ```
+  /// let params = CheckForUpdateParams(accessToken: "your_access_token")
+  /// checkForUpdate(params: params) { result in
+  ///     switch result {
+  ///     case .success(let releaseInfo):
+  ///       if let releaseInfo = releaseInfo.update {
+  ///         print("Update found: \(releaseInfo)")
+  ///       } else {
+  ///         print("Already up to date")
+  ///       }
+  ///     case .failure(let error):
+  ///         print("Error checking for update: \(error)")
+  ///     }
+  /// }
+  /// ```
+  public func checkForUpdate(params: CheckForUpdateParams,
+                             completion: @escaping ((Result<UpdateCheckResponse, Error>) -> Void)) {
+    getUpdatesFromBackend(params: params, completion: completion)
+  }
+
+  /// Obtain a URL to install an IPA
+  /// - Parameter plistUrl: The URL to the plist containing the IPA information
+  /// - Returns: a URL ready to install the IPA using Itunes Services
+  public static func buildUrlForInstall(_ plistUrl: String) -> URL? {
+    guard var components = URLComponents(string: "itms-services://") else {
+      return nil
+    }
+    components.queryItems = [
+      URLQueryItem(name: "action", value: "download-manifest"),
+      URLQueryItem(name: "url", value: plistUrl)
+    ]
+    return components.url
+  }
+
+  // MARK: - Private
+  private lazy var session = URLSession(configuration: URLSessionConfiguration.ephemeral)
```

Comment:
`m`: Add an initializer with an ephemeral session as the default value. We want this to be injectable for unit tests